server/admin_management/api/app/services/user_service.py

```diff
@@ -19,32 +19,81 @@ def get_user_by_domain_and_name(self, user_type_code: str, user_name: str):
     def get_user_by_cognito_user_id(self, cognito_user_id: str):
         return self.user_repo.get_user_by_cognito_user_id(cognito_user_id)
 
+    def get_user_by_domain_and_guid(self, user_type_code: str, user_guid: str):
+        return self.user_repo.get_user_by_domain_and_guid(user_type_code, user_guid)
+
     def get_users(self):
         return self.user_repo.get_users()
 
-    def find_or_create(self, user_type_code: str, user_name: str, requester: str):
+    def find_or_create(
+        self, user_type_code: str, user_name: str, user_guid: str, requester: str
+    ):
         LOGGER.debug(
             f"Request for finding or creating a user with user_type: {user_type_code}, "
             + f"user_name: {user_name}."
         )
 
-        fam_user = self.get_user_by_domain_and_name(user_type_code, user_name)
+        fam_user = self.get_user_by_domain_and_guid(user_type_code, user_guid)
         if not fam_user:
-            request_user = FamUserDto(
-                **{
-                    "user_type_code": user_type_code,
-                    "user_name": user_name,
-                    "create_user": requester,
-                }
+            fam_user_by_domain_and_name = self.get_user_by_domain_and_name(
+                user_type_code, user_name
             )
+            # if not found user by user_guid, but found user by domain and name
+            # in the case of historical FAM user that has no user_guid stored
+            # add their user_guid
+            if (
+                fam_user_by_domain_and_name
+                and not fam_user_by_domain_and_name.user_guid
```

Comment:
ideally, if we did a one-time load to add the user_guid for all existing users, we won't need this logic anymore. But it's also not hurt to have it